media/vp8_ivf.ksy

```diff
@@ -49,23 +50,62 @@ seq:
 
 ## payload start
   - id: image_data
-    type: blocks
+    type: frames
     repeat: expr
     repeat-expr: num_frames
 ## payload end
 
 ## type definitions
 types:
-  blocks:
+  frames:
     seq:
       - id: entries
-        type: block
-  block:
+        type: frame
+  frame:
     seq:
       - id: len_frame
         doc: size of the frame data
         type: u4
       - id: timestamp
         type: u8
-      - id: framedata
-        size: len_frame
+      - id: is_interframe
+        doc: A 1-bit frame type (0 for key frames, 1 for interframes).
+        type: b1 
+      - id: version_number
+        doc: >
+          A 3-bit version number (0 - 3 are defined as four different 
+          profiles with different decoding complexity; other values may be 
+          defined for future variants of the VP8 data format).
+        type: b3
+      - id: flag_show_frame
+        doc: >
+          A 1-bit show_frame flag (0 when current frame is not for display,
+          1 when current frame is for display).
+        type: b1
+      - id: first_data_partition_size
+        doc: >
+          A 19-bit field containing the size of the first data partition in
+          bytes.
+        type: b19
+      - id: keyframe_hdr
+        type: keyframe_tag
+        if: not is_interframe
+      - id: compressed_keyframe
+        size: len_frame - 10
+        if: not is_interframe
+      - id: compressed_interframe
+        size: len_frame - 3
+        if: is_interframe
```

Comment:
What are these constants `10` and `3` that you're subtracting? What fields they are sizes of?

You know, calculating the sizes manually like that with subtraction is almost always a bad idea, and it can be usually done better with [substreams](https://doc.kaitai.io/user_guide.html#stream).

Reply:
@generalmimon Every frame has a 3-byte frame tag: https://tools.ietf.org/html/rfc6386#section-9.1 ,if it is a keyframe (defined by the first bit of that tag), then another 7 bytes are used for further information. On the same page: "...For key frames, the frame tag is followed by a further 7 bytes of uncompressed data, ..."
As they are fixed in the specification I see no case where this could be troublesome. May be a comment?